src/modules/iotjs_module_buffer.c

```diff
@@ -411,12 +409,12 @@ JHANDLER_FUNCTION(Slice) {
   int length = end - start;
   IOTJS_ASSERT(length >= 0);
 
-  iotjs_jval_t jnew_buffer = iotjs_bufferwrap_create_buffer(length);
+  iotjs_jval_t jnew_buffer = iotjs_bufferwrap_create_buffer((size_t)length);
   iotjs_bufferwrap_t* new_buffer_wrap =
       iotjs_bufferwrap_from_jbuffer(&jnew_buffer);
   iotjs_bufferwrap_copy_internal(new_buffer_wrap,
-                                 iotjs_bufferwrap_buffer(buffer_wrap), start,
-                                 end, 0);
+                                 iotjs_bufferwrap_buffer(buffer_wrap),
+                                 (size_t)start, (size_t)end, 0);
```

Comment:
@chokobole @LaszloLango 
Our documentation of [`Buffer`](https://github.com/Samsung/iotjs/blob/master/docs/api/IoT.js-API-Buffer.md#bufcopytargetbuffertargetstartsourcestartsourceend) says `targetStart`, `sourceStart`, and `sourceEnd` as `Number`. According to document, the type of `start` should be `double`. But I think the documentation is wrong since we will not support such a big range. So first, we have to correct the API document `Number` to `Integer`.  Then, second, we need to define what we will do if the user gives out of Integer range. node.js seems to cast the double number into int64_t if it is not SMI. If I understand correctly, and we agree to follow the style,  finally we can use `int64_t` as type of `start`.